xla/hlo/evaluator/hlo_evaluator.cc

```diff
@@ -255,6 +255,9 @@ struct DynamicOrStaticInteger {
   }
 };
 
+static_assert(std::is_trivially_copyable_v<DynamicOrStaticInteger>,
+              "DynamicOrStaticInteger is expected to be trivially copyable");
```

Comment:
I personally don't think it's worth static_asserting these types are trivially copyable. Without the context of this PR, it's non-obvious to someone reading the code why there is a static assert, and in the very-unlikely case someone does add a non-trivially-copyable field to these types, the only downside will be negligible lower performance since we don't std::move it.